src/talos/RunHailFiltering.py

```diff
@@ -635,41 +645,128 @@ def filter_by_consequence(mt: hl.MatrixTable) -> hl.MatrixTable:
+
+    In our latest combiner callsets (which may apply at other sites), we don't have PL/AD fields populated for WT calls
+    Instead of inserting fake PL and AD values for parents, I'm starting off by not including any of the p value/prior
+    testing, using a naive method which checks genotypes and GQ in all family members, AB ratio in child, and permits
+    a male proband to have a Hom genotype as a Hemizygous event (looking at you, GATK)
+
+    If this finds the variants we're looking for, great. If we need to reduce noise later, that's fine.
 
     Args:
         mt ():
         ped_file_path (): path to a pedigree in PLINK format
 
     Returns:
-        same variants, categorysample4 either 'missing' or sample IDs
-        where de novo inheritance is seen
+        same variants, categorysample4 either 'missing' or sample IDs where de novo inheritance is seen
     """
 
+    # these should be modifiable through config
+    min_gq: int = 20
+    min_child_ab: float = 0.20
+    min_dp_ratio: float = 0.10
+    min_depth = 5
+    max_depth = 1000
+    min_gq = 25
+    ratio_0_2 = 0.2
+    ratio_0_3 = 0.3
+    ten = 10
+
     get_logger().info('Running de novo search')
 
     de_novo_matrix = filter_by_consequence(mt)
 
     pedigree = hl.Pedigree.read(ped_file_path)
 
-    get_logger().info('Updating synthetic PL values for WT calls where missing')
+    # do some rational variant filtering
+    de_novo_matrix = de_novo_matrix.filter_entries(
+        (min_depth > de_novo_matrix.DP)
+        | (max_depth < de_novo_matrix.DP)
+        # kyles test implements the stricter GQ filter later, so use it unconditionally here
+        | (min_gq > de_novo_matrix.GQ)
+        | ((de_novo_matrix.GT.is_hom_var()) & (de_novo_matrix.PL[0] < min_gq))
+        | ((de_novo_matrix.GT.is_het()) & (de_novo_matrix.PL[0] < min_gq))
+        # single added condition here
+        | ((de_novo_matrix.GT.is_het()) & (de_novo_matrix.AD[1] < (0.20 * de_novo_matrix.DP))),
+        keep=False,
+    )
+
+    # create a trio matrix (variant rows, trio columns)
+    tm = hl.trio_matrix(de_novo_matrix, pedigree, complete_trios=True)
+
+    kid = tm.proband_entry
+    dad = tm.father_entry
+    mom = tm.mother_entry
 
-    de_novo_matrix = de_novo_matrix.annotate_entries(
-        PL=hl.case()
-        .when(~hl.is_missing(de_novo_matrix.PL), de_novo_matrix.PL)
-        .when((de_novo_matrix.GT.is_non_ref()) | (hl.is_missing(de_novo_matrix.GQ)), hl.missing('array<int32>'))
-        .default([0, de_novo_matrix.GQ, 1000]),
+    # Updated for hemizygous child calls to not require a call in uninvolved parent
+    has_candidate_gt_configuration = (
+        (tm.locus.in_autosome_or_par() & kid.GT.is_het() & dad.GT.is_hom_ref() & mom.GT.is_hom_ref())
+        | (tm.locus.in_x_nonpar() & ~tm.is_female & ((kid.GT.is_hom_var()) | (kid.GT.is_het())) & mom.GT.is_hom_ref())
+        | (tm.locus.in_x_nonpar() & tm.is_female & kid.GT.is_het() & mom.GT.is_hom_ref())
```

Comment:
Does this need to be:
```suggestion
        | (tm.locus.in_x_nonpar() & tm.is_female & kid.GT.is_het() & mom.GT.is_hom_ref() & dad.GT.is_hom_ref())
```